src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/dateHelpers.ts

```diff
@@ -0,0 +1,26 @@
+import moment from 'moment';
+import { DateFlags } from 'src/types';
+
+/*
+  Creates a specific date based on different flags (DatepickerRestrictionFlags)
+  Returns moment.Moment or null
+*/
+export function getFlagBasedDate(flag: DateFlags): string | undefined {
+  if (flag === DateFlags.Today) {
+    return moment()
+      .set('hour', 12)
```

Comment:
```suggestion
      .set('hour', 0)
```

Or do you have reason to put this to 12? Wouldn't this cause problem for instance when we set minDate: "today" and the user selects the current when the time actually is before 12?

Reply:
Oh, then we might have an issue with maxDate if we set it to 0. Can we potentially end up in a weird state here? 🤔

Reply:
The only reason is because I saw the same code in handleDataChangeWrapper(...) and it seemed like a good idea at the time. Ah, I understand your example now, sure, thats a good fix. I thought 12 was default values, because it was used in the method mentioned above.

Reply:
Ah, I got it now. Scratch what I said, leaving it at 12 sounds like a solid plan. What I saw a bug at was that when validating in `isValidDate` in the compoent we don't set the hour and the minut which caused some issues. I pushed a fix for this.